cli/util/file_watcher.rs

```diff
@@ -120,6 +148,10 @@ pub struct WatcherCommunicator {
 
   /// Send a message to force a restart.
   restart_tx: tokio::sync::mpsc::UnboundedSender<()>,
+
+  restart_mode: Arc<Mutex<WatcherRestartMode>>,
```

Comment:
Would it be better if we put an Arc around the entire WatcherCommunicator? (or actually looks like we might be able to do Rc?)

Reply:
I'm not sure it would help much as `changed_paths_rx` requires an explicit function call to create a "clone" of it. `restart_mode` is the only field that requires internal mutability.

Reply:
I just tried it out and it works to remove the `Clone` implementation then just pass around an `Arc<WatcherCommunicator>` everywhere. It's much more clear that it's the same instance being passed around when it's in an `Arc`.